tests/tqueue_int/tqueue_int.c

```diff
@@ -406,21 +413,28 @@ static void test_and_terminate_chaos_test(TQUEUE_CHAOS_TEST_THANDLE_CONTEXT *tes
         int64_t last_successful_pop_count = interlocked_add_64(&test_context->successful_pop_count, 0);
 
         ThreadAPI_Sleep(TEST_CHECK_PERIOD);
-        
+
+        INTERLOCKED_HL_RESULT wait_result;
+
         // Ensures that the tests below for current_successful_push_count > last_successful_push_count and/or current_successful_pop_count > last_successful_pop_count
         // do not fail due to multi-threading synchronization issues
-        ASSERT_ARE_EQUAL(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_OK, InterlockedHL_WaitForNotValue64(&test_context->successful_push_count, last_successful_push_count, INT_MAX));
-        ASSERT_ARE_EQUAL(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_OK, InterlockedHL_WaitForNotValue64(&test_context->successful_pop_count, last_successful_pop_count, INT_MAX));
+        do
+        {
+            wait_result = InterlockedHL_WaitForNotValue64(&test_context->successful_push_count, last_successful_push_count, PROGRESS_WAIT_TIME);
+            ASSERT_IS_TRUE((wait_result == INTERLOCKED_HL_OK) || (wait_result == INTERLOCKED_HL_TIMEOUT));
+        } while ((interlocked_add(&test_context->active_chaos_threads, 0) > 0) && (wait_result == INTERLOCKED_HL_TIMEOUT));
```

Comment:
>&& (wait_result == INTERLOCKED_HL_TIMEOUT) [](http://example.com/codeflow?start=79&length=42)

doesn't really matter especially here, but could reverse the condition (check wait_result first) and skip the interlocked_add in the common case

Reply:
Done